src/Aggregation/BucketScript.php

```diff
@@ -13,17 +11,12 @@ class BucketScript extends AbstractAggregation implements GapPolicyInterface
 {
     use Traits\GapPolicyTrait;
 
-    public function __construct(string $name, ?array $bucketsPath = null, ?string $script = null)
+    public function __construct(string $name, array $bucketsPath, string $script)
     {
         parent::__construct($name);
 
-        if (null !== $bucketsPath) {
```

Comment:
This one we did not deprecate. 

One thing I've seen in the past is that an object was created when the bucketPath did not exist yet and was set at a later stage (just an example). Is this a use case we want to support or on this scenario we except the object to be created when all info is available?

Reply:
Yeah, I would deprecate them in `7.x`.

When working with ES, these parameters are required:

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-bucket-script-aggregation.html
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-bucket-selector-aggregation.html

Since they are required, I think it makes sense to not be able to create a `BucketScript` instance with invalid data.

IMHO we should avoid supporting those scenarios where these classes can contain invalid data (in some of them we can even mark them as [immutables](https://phpstan.org/writing-php-code/phpdocs-basics#immutable-classes)), it simplifies the code and ease the maintenance since these object would be always in a valid state.

Reply:
Lets call this a decision that all required params have to be set when creating an object.

Can you open a PR against 7.x with a deprecation?